nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/PublishGCPubSub.java

```diff
@@ -162,7 +162,7 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
 
                     ApiFuture<String> messageIdFuture = publisher.publish(message);
 
-                    while (messageIdFuture.isDone()) {
+                    while (!messageIdFuture.isDone()) {

```

Comment:
@zenfenan is there a reason to have this loop at all? The next line calls `messageIdFuture.get()`, which is a blocking call. So we should just remove this entire loop, as it's adding up to a half-second delay for each FlowFile. Or am I missing something?

Reply:
@markap14 That is true. I understand that `.get()` is a blocking call but when I was implementing this processor and tested this without this waiting while loop, I saw a few messages weren't getting published. I wasn't sure if it is something on the GCP side. I am not sure if it is still an issue and we can drop that piece of code since I don't have available GCP resources to try. Maybe @pvillard31 will have some idea.